### PR TITLE
Don't throw an error if sourceMap is null/undefined

### PR DESCRIPTION
**Pull request by jiaruiz717**
fix iTwin/itwinjs-backlog#1123

sourceMap could be undefined/null in some cases. ignore it instead of throwing an error

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

File: core/backend/src/Material.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export class RenderMaterialElement extends DefinitionElement {
       if (typeof mapName !== "string")
         continue;
       const sourceMap = sourceProps.jsonProperties.materialAssets.renderMaterial.Map[mapName];
-      if (!Id64.isValid(sourceMap.TextureId) || sourceMap.TextureId === undefined)
+      if (!sourceMap || !Id64.isValid(sourceMap.TextureId) || sourceMap.TextureId === undefined)
```
[the remark this answers is not in the document]
Comment (jiaruiz717) [reply core/backend/src/Material.ts:196]: added

**Comment by nick4598** [comment]
Can you look at the tests added by the PR in the workitem, and add to those potentially making sure that the null gets persisted?

File: core/backend/src/test/standalone/RenderMaterialElement.test.ts [hunk 2]
```diff
@@ -358,6 +358,13 @@ describe("RenderMaterialElement", () => {
         UnknownTexture: { TextureId: "CLONED" },
         NoTextureId: { OtherProp: 1 },
       });
+
+      jsonProps.materialAssets.renderMaterial.Map = {Pattern: undefined};
+      material.update();
+      // eslint-disable-next-line @typescript-eslint/dot-notation
+      RenderMaterialElement["onCloned"](context, sourceProps, targetProps);
+      // keep the sourceMap the same in targetProps
+      expect(targetProps.jsonProperties?.materialAssets?.renderMaterial?.Map?.Pattern).to.be.undefined;
```
Comment (nick4598) [inline core/backend/src/test/standalone/RenderMaterialElement.test.ts:367]: Couldn't this return undefined even if it didn't make it to Pattern?

I.e. if anything in the chain (JsonProperties, materialAssets, rendermaterial, map) is undefined it returns undefined.

File: core/backend/src/test/standalone/RenderMaterialElement.test.ts [hunk 3]
```diff
@@ -358,6 +358,13 @@ describe("RenderMaterialElement", () => {
         UnknownTexture: { TextureId: "CLONED" },
         NoTextureId: { OtherProp: 1 },
       });
+
+      jsonProps.materialAssets.renderMaterial.Map = {Pattern: undefined};
```
Comment (nick4598) [inline core/backend/src/test/standalone/RenderMaterialElement.test.ts:362]: Can you check Pattern: null as well?

**Pull request merged**